File: scripts/create.js
const mysql = require('mysql');
const fs = require('fs');

// Configuration for the MySQL connection
const dbConfig = {
  host: 'localhost',
  user: 'root',
  password: 'root',
  multipleStatements: true
};

const connection = mysql.createConnection(dbConfig);

// Function to execute the SQL script
function runSqlScript() {
  const scriptFilePath = './sql/database_init.sql';
  fs.readFile(scriptFilePath, 'utf8', (err, sqlScript) => {
    if (err) {
      console.error('Error reading SQL script file:', err);
      return;
    }

    connection.query(sqlScript, (queryErr, results, fields) => {
      if (queryErr) {
        console.error('Error executing SQL script:', queryErr);
        return;
      }

      console.log('SQL script executed successfully.');

      // Close the database connection
      closeConnection();
    });
  });
}

// Function to close the MySQL connection
function closeConnection() {
  connection.end((endErr) => {
    if (endErr) {
      console.error('Error closing MySQL connection:', endErr);
    } else {
      console.log('MySQL connection closed.');
    }
  });
}

// Initialize the database connection and execute the SQL script
connection.connect((connErr) => {
  if (connErr) {
    console.error('Error connecting to MySQL:', connErr);
    return;
  }

  runSqlScript();
});

File: create.py
import pymysql
import os

# Configuration for the MySQL connection
db_config = {
    'host': 'localhost',
    'user': 'root',
    'password': 'root',
    'database': 'joy_of_painting'
}

# Connect without specifying the database first
connection = pymysql.connect(
    host=db_config['host'],
    user=db_config['user'],
    password=db_config['password'],
    database=db_config['database']
)

# Function to create tables
def create_tables():
    script_file_path = './sql/database_init.sql'
    if os.path.exists(script_file_path):
        with open(script_file_path, 'r') as file:
            sql_script = file.read()
            statements = sql_script.split(';')
            with connection.cursor() as cursor:
                for statement in statements:
                    if statement.strip():
                        cursor.execute(statement)
                connection.commit()
                print('Tables created successfully.')

# Function to close the MySQL connection
def close_connection():
    connection.close()
    print('MySQL connection closed.')

# Initialize the database connection and execute the SQL script
try:
    create_tables()
finally:
    close_connection()
